agent/consul/state/catalog.go

```diff
@@ -1090,6 +1090,33 @@ func assignServiceVirtualIP(tx WriteTxn, idx uint64, psn structs.PeeredServiceNa
 	return result.String(), nil
 }
 
+func (s *Store) AssignManualVirtualIPs(idx uint64, psn structs.PeeredServiceName, ips []string) error {
```

Comment:
This looks good, but I'm starting to wonder if we should be extra-careful and remove old VIPs if there happens to be a conflict when performing this upsert. Do you have any thoughts about this? It's extra effort, but it might save us from a nasty bug where consul-k8s doesn't clean up old VIPs.

Reply:
Updated the PR to add this behavior - I think k8s not cleaning up old VIPs is a real concern, though allowing services to "steal" manual VIP entries from others feels a bit off, especially when considering ACLs or namespaces/partitions. Not sure if there's a cleaner solution though - we really want to avoid the case where multiple upstreams with shared manual VIPs end up on the same transparent proxy setup, and checking for duplicates globally seems like the only way.